nilearn/decoding/searchlight.py

```diff
@@ -353,7 +395,55 @@ def fit(self, imgs, y, groups=None):
             self.n_jobs,
             self.verbose,
         )
-        scores_3D = np.zeros(process_mask.shape)
-        scores_3D[process_mask] = scores
-        self.scores_ = scores_3D
+        self.masked_scores_ = scores
+        self.scores_ = np.zeros(process_mask.shape)
+        self.scores_[np.where(process_mask)] = scores
         return self
+
+    @property
+    def scores_img_(self):
+        """Convert the 3D scores array into a NIfTI image."""
+        if self.scores_ is None:
+            raise ValueError(
+                "The model has not been fitted yet. Call `fit()` "
+                "before accessing `scores_img_`."
+            )
+        return new_img_like(self.mask_img, self.scores_)
+
+    def transform(self, imgs):
+        """Apply the fitted searchlight on new images."""
+        if self.process_mask_ is None or self.scores_ is None:
+            raise ValueError(
+                "You must fit the model before calling `transform()`."
+            )
+
+        imgs = check_niimg_4d(imgs)
+
+        X, A = _apply_mask_and_get_affinity(
+            np.asarray(np.where(self.process_mask_)).T,
+            imgs,
+            self.radius,
+            True,
+            mask_img=self.mask_img,
+        )
+
+        estimator = self.estimator
```

Comment:
```suggestion
        estimator = self.estimator
        if estimator == "svc":
            estimator = ESTIMATOR_CATALOG[estimator](dual=True)
        elif isinstance(estimator, str):
            estimator = ESTIMATOR_CATALOG[estimator]()
```

need to do the same in as in `fit` in case the use passed `"svc"`

Reply:
that should fix the error